.chloggen/3950.yaml

```diff
@@ -0,0 +1,21 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. collector, target allocator, auto-instrumentation, opamp, github action)
+component: collector
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Ports are being duplicated for ContainerSpec when user provides them.
+
+# One or more tracking issues related to the change
+issues: [3950]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  This fixes a bug where ports are being duplicated for ContainerSpec when user provides them.
```

Comment:
This patch improves container port collisions when the user defines additional ports in the CR.